frontend/model/notifications/templates.js

```diff
@@ -1,133 +1,100 @@
-import L from '@view-utils/translations.js'
-import { timeSince } from '@utils/time.js'
+import type {
+  NewProposalType,
+  NotificationTemplate
+} from './types.flow.js'
 
-type TemplateArguments = {
-  timestamp: number;
-  data: Object;
-}
+import L from '~/frontend/views/utils/translations.js'
+import { monthName } from '~/frontend/utils/time.js'
 
-type TemplateDescription = {
-  avatar: string;
-  icon: string;
-  level: 'danger' | 'info';
-  body: string;
-  linkTo: string;
-  date: string;
-  read: boolean;
-}
+// Note: this escaping is not intended as a protection against XSS.
+// It is only done to enable correct rendering of special characters in usernames.
+// To guard against XSS when rendering usernames, use the `v-safe-html` directive.
+const escapeForHtml = (text) => text.replace(/[<>&]/g, '\\$&')
+const strong = (text) => `<strong>${escapeForHtml(text)}</strong>`
 
-// Notification templates based on issue #663
-// TODO/REVIEW - list might be stale / incomplete.
+export default ({
+  CONTRIBUTION_REMINDER (data: { monthstamp: string }) {
+    return {
+      // REVIEW @mmbotelho - Rename contributions to payments?
+      // TODO: Translate `month` when English is not the current language.
+      body: L('Do not forget to send your {month} contributions.', {
+        month: strong(monthName(data.monthstamp))
+      }),
+      icon: 'coins',
+      level: 'info',
+      linkTo: '/payments',
+      scope: 'user'
+    }
+  },
+  INCOME_DETAILS_OLD (data: { months: number }) {
+    return {
+      body: L("You haven't updated your income details in more than {months} months. Would you like to review them now?", {
+        // Avoid displaying decimals.
+        months: Math.floor(data.months)
+      }),
+      icon: 'coins',
+      level: 'info',
+      linkTo: '/contributions/TODO-LINK-MODAL',
+      scope: 'user'
+    }
+  },
+  MEMBER_ADDED (data: { groupID: string, username: string }) {
+    return {
+      body: L('The group has a new member. Say hi to {name}!', {
+        name: strong(data.username)
+      }),
```

Comment:
Can you double-check that this is safe?

I know that the `i18n` component uses `dompurify` (see `views/utils/translations.js`), but I don't think the `L` function does. 

So if we're passing in user provided data like `data.username` here, is this a potential issue? Or was the username already sanitized? Or...?

Reply:
I think `data.username` here gets sanitized later by using the `v-safe-html` directive:

https://github.com/okTurtles/group-income/blob/188a33d9a3b666881f9fe58c5e1029a70e6bd436/frontend/views/containers/notifications/NotificationList.vue#L24

Reply:
OK good. @snowteamer, then can you add a comment here letting people know that they must include this in the page using `v-safe-html` to protect against XSS?

Reply:
I have already included a similar comment above `strong() `definition. Maybe it was not the best place. Would a reference to that comment be fine, or do you think it's better to duplicate or move that comment to here?

Reply:
Oh! I forgot about that. Sorry! Resolving.